auth/oidc.go

```diff
@@ -141,88 +199,86 @@ func (op *OIDCProvider) InitOIDCClient() error {
 		return base.RedactErrorf("Issuer not defined for OpenID Connect provider %+v", base.UD(op))
 	}
 
-	config, shouldSyncConfig, err := op.DiscoverConfig()
-	if err != nil || config == nil {
+	ctx := context.Background()
+	verifier, endpoint, err := op.DiscoverConfig(ctx)
+	if err != nil || verifier == nil {
 		return pkgerrors.Wrap(err, "unable to discover config")
 	}
 
-	clientCredentials := oidc.ClientCredentials{
-		ID: *op.ClientID,
-	}
-	if op.ValidationKey != nil {
-		clientCredentials.Secret = *op.ValidationKey
+	config := oauth2.Config{
+		ClientID:    *op.ClientID,
+		Endpoint:    *endpoint,
+		RedirectURL: *op.CallbackURL,
 	}
 
-	clientConfig := oidc.ClientConfig{
-		ProviderConfig: *config,
-		Credentials:    clientCredentials,
-		RedirectURL:    *op.CallbackURL,
+	if op.ValidationKey != nil {
+		config.ClientSecret = *op.ValidationKey
 	}
 
 	if op.Scope != nil || len(op.Scope) > 0 {
-		clientConfig.Scope = op.Scope
+		config.Scopes = op.Scope
 	} else {
-		clientConfig.Scope = []string{"openid", "email"}
-	}
-
-	op.OIDCClient, err = oidc.NewClient(clientConfig)
-	if err != nil {
-		return err
-	}
-
-	// Start process for ongoing sync of the provider config
-	if shouldSyncConfig {
-		op.OIDCClient.SyncProviderConfig(op.Issuer)
-	} else {
-		base.Infof(base.KeyAuth, "Not synchronizing provider config for issuer %s...", base.UD(op.Issuer))
+		config.Scopes = []string{oidc.ScopeOpenID, "email"}
 	}
 
 	// Initialize the prefix for users created for this provider
 	if err = op.InitUserPrefix(); err != nil {
 		return err
 	}
 
+	op.client = &OIDCClient{Config: config, Context: ctx, Verifier: verifier}
+
 	return nil
 }
 
-func (op *OIDCProvider) DiscoverConfig() (config *oidc.ProviderConfig, shouldSync bool, err error) {
-
+func (op *OIDCProvider) DiscoverConfig(ctx context.Context) (verifier *oidc.IDTokenVerifier, endpoint *oauth2.Endpoint, err error) {
```

Comment:
It looks like we're still passing the background context down the stack from InitOIDCClient to here.  Since this is just the background context, I think it's sufficient to just pass the background context directly to the context-aware functions being called in DiscoverConfig, and remove the parameter.

Reply:
Removed background context parameter from DiscoverConfig and added changes to just pass background context directly to the context-aware function called directly in DiscoverConfig.